pkg/relay/processmanager.go

```diff
@@ -37,11 +41,16 @@ type ProcessManager struct {
 	m ProcessManagerMetrics
 }
 
-func NewProcessManager(l log.Logger, verifySize, storeSize uint) *ProcessManager {
-	rm := &ProcessManager{
-		l:           l,
-		LastRegTime: make(map[string]uint64),
+func NewProcessManager(l log.Logger, verifySize, storeSize uint, registrationCacheSize int) *ProcessManager {
```

Comment:
Why don't we return `(*ProcessManager, error)`?

Reply:
In general constructors shouldn't return an error ;) but yeah... you're right, I'll change that